criu/files.c

```diff
@@ -1372,11 +1390,15 @@ int restore_fs(struct pstree_item *me)
 	 * Now do chroot/chdir. Chroot goes first as it calls chdir into
 	 * dd_root so we'd need to fix chdir after it anyway.
 	 */
-
-	ret = fchroot(dd_root);
-	if (ret < 0) {
-		pr_perror("Can't change root");
-		goto out;
+	if (need_chroot(dd_root)) {
```

Comment:
We don't need to do any specific actions if opts.unprivileged isn't set. The code can look like this:
```
if (opts.unprivileged)
  check_root(dd_root);
else {
		ret = fchroot(dd_root);
		if (ret < 0) {
			pr_perror("Can't change root");
			goto out;
		}
}
```

Reply:
Fixed, the check will only be done in unprivileged mode.